tests/integ/test_simplifier_suite.py

```diff
@@ -995,6 +996,46 @@ def test_rename_to_existing_column_column(session):
     Utils.check_answer(df4, [Row(3, 3, 1)])
 
 
+@pytest.mark.parametrize(
+    "operation,simplified_query",
```

Comment:
If I use sql expression, does it flatten, for instance `df.filter(sql_expr('a > 1')).select(...)`?
It shouldn't because we can't detect the dependent columns of a sql text.

Reply:
Good catch! I fixed and added a test case.